src/ts.rs

```diff
@@ -19,6 +21,42 @@ pub mod internal {
         TypeDefs,
     };
 
+    /// The configuration for the generator
+    #[derive(Default)]
+    pub struct ExportConfiguration {
```

Comment:
This shouldn't be nested in `pub mod internal` just in the root of this file. It's not internal as it's used by other functions that are not within the `internal` module.